compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -410,6 +410,7 @@ private sealed trait YSettings:
   // Experimental language features
   val YnoKindPolymorphism: Setting[Boolean] = BooleanSetting("-Yno-kind-polymorphism", "Disable kind polymorphism.")
   val YexplicitNulls: Setting[Boolean] = BooleanSetting("-Yexplicit-nulls", "Make reference types non-nullable. Nullable types can be expressed with unions: e.g. String|Null.")
+  val YflexibleTypes: Setting[Boolean] = BooleanSetting("-Yflexible-types", "Make Java return types and parameter types use flexible types, which have a nullable lower bound and non-null upper bound.")
```

Comment:
Should we consider making this on by default? So the setting would be `YnoFlexibleTypes`.

Reply:
After discussing with Martin, we agree to make this feature on by default. One thing we need be careful is that with this PR, the dotty compiler will be compiled with both flexible types and unsafe nulls. The goal is to remove unsafe nulls after this PR is merged.

Reply:
I tried to make it by default but there are some work to do. Since the current explicit nulls tests are based on nullable types from Java, we need to find a way to handle these tests.